tools/VscPreviewExtension/src/dfmHttpClient.ts

```diff
@@ -1,26 +1,31 @@
 // Copyright (c) Microsoft. All rights reserved. Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-import { default as Axios, AxiosResponse } from 'axios'
+import { default as Axios, AxiosResponse } from 'axios';
 
-import { DfmServiceResult } from './dfmServiceResult';
+import * as ConstVariable from "./ConstVariable";
 
 export class DfmHttpClient {
-    // TODO: make the urlPrefix configurable
-    private urlPrefix = "http://localhost:4001";
+    private static urlPrefix = "http://localhost:";
 
-    async sendPostRequestAsync(command: String, content: String): Promise<DfmServiceResult> {
-        let promise = Axios.post(this.urlPrefix, {
+    static async sendPostRequest(port: string, command, content = null, workspacePath = null, relativePath = null, writeTempPreviewFile = false, previewFilePath = null, pageRefreshJsFilePath = null, builtHtmlPath = null): Promise<AxiosResponse> {
+        let promise = Axios.post(this.urlPrefix + port, {
             name: command,
-            documentation: content
+            markdownContent: content,
+            previewFilePath: previewFilePath,
+            workspacePath: workspacePath,
+            relativePath: relativePath,
+            writeTempPreviewFile: writeTempPreviewFile,
+            builtHtmlPath: builtHtmlPath,
+            pageRefreshJsFilePath: pageRefreshJsFilePath
         });
 
-        let response : AxiosResponse;
+        let response: AxiosResponse;
         try {
             response = await promise;
         } catch (err) {
             let record = err.response;
             if (!record) {
-                throw new Error(err)
+                throw new Error(ConstVariable.noServiceErrorMessage);
```

Comment:
previewCore.ts line 25 and tokentree.ts line 24

---
In reply to: [111545489](https://github.com/dotnet/docfx/pull/1547#discussion_r111545489) [](ancestors = 111545489)